var/spack/repos/builtin/packages/uncrustify/package.py

```diff
@@ -10,12 +10,30 @@ class Uncrustify(Package):
     """Source Code Beautifier for C, C++, C#, ObjectiveC, Java, and others."""
 
     homepage = "http://uncrustify.sourceforge.net/"
-    url      = "http://downloads.sourceforge.net/project/uncrustify/uncrustify/uncrustify-0.61/uncrustify-0.61.tar.gz"
-
-    version('0.67', sha256='54f15c8ebddef120522db21f38fac1dd3b0a285fbf60a8b71f9e333e96cf6ddc')
+    git      = "https://github.com/uncrustify/uncrustify"
+    url      = "https://sourceforge.net/projects/uncrustify/files/uncrustify/uncrustify-0.69/uncrustify-0.69.tar.gz"
+
+    maintainers = ['gmaurel']
+
+    version('master', branch='master')
+    version('0.74', commit='62048b')
+    version('0.73', commit='25b765')
+    version('0.72', commit='1d3d8f')
+    version('0.71', commit='64d82f')
+    version('0.70', commit='51f64d')
+    version('0.69', commit='a7a8fb')
+    version('0.68', commit='86bc34')
+    version('0.67', commit='00321a')
+    version('0.66', commit='80f549')
+    version('0.65', commit='905676')
+    version('0.64', commit='1d7d97')
+    version('0.63', commit='44ce0f')
+    version('0.62', commit='5987f2')
```

Comment:
I wasn't aware you could specify a shortened commit hash but as long as you can `spack stage uncrustify@0.62` then this is fine.

Reply:
According to the [Docs for Git-Commits](https://spack.readthedocs.io/en/latest/packaging_guide.html#git) the shortened hash version is supported. Also I tested `spack install uncrustify@0.62` and it works as expected. Haven't tested staging explicitely but I wouldn't expect it to behave differently than install?
But I can test it the next tima I am at my wokstation, if you wish?

Reply:
Also tested `spack stage` now and works as expected.